src/Acuminator/Acuminator.Analyzers/StaticAnalysis/PublicClassXmlComment/PublicClassXMLCommentAnalyzer.cs

```diff
@@ -0,0 +1,172 @@
+﻿using Acuminator.Utilities;
+using Acuminator.Utilities.DiagnosticSuppression;
+using Acuminator.Utilities.Roslyn.Constants;
+using Acuminator.Utilities.Roslyn.Semantic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using static Acuminator.Analyzers.StaticAnalysis.PublicClassXmlComment.PublicClassXmlCommentFix;
+
+namespace Acuminator.Analyzers.StaticAnalysis.PublicClassXmlComment
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class PublicClassXmlCommentAnalyzer : PXDiagnosticAnalyzer
+    {
+		private static readonly string[] _xmlCommentSummarySeparators = { SyntaxFactory.DocumentationComment().ToFullString() };
+
+		public static readonly string XmlCommentSummaryTag = SyntaxFactory.XmlSummaryElement().StartTag.Name.ToFullString();
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+            ImmutableArray.Create(Descriptors.PX1007_PublicClassXmlComment);
+
+        public PublicClassXmlCommentAnalyzer(CodeAnalysisSettings codeAnalysisSettings)
+            : base(codeAnalysisSettings)
+        {
+        }
+
+		public PublicClassXmlCommentAnalyzer()
+			: this(null)
+		{
+		}
+
+        internal override void AnalyzeCompilation(CompilationStartAnalysisContext compilationStartContext, PXContext pxContext)
+        {
+            compilationStartContext.RegisterSyntaxNodeAction(AnalyzeClassDeclaration, SyntaxKind.ClassDeclaration);
+        }
+
+        private void AnalyzeClassDeclaration(SyntaxNodeAnalysisContext syntaxContext)
+        {
+            syntaxContext.CancellationToken.ThrowIfCancellationRequested();
+
+            if (!(syntaxContext.Node is ClassDeclarationSyntax classDeclarationSyntax))
+            {
+                return;
+            }
+
+            var isPublicClass = classDeclarationSyntax.Modifiers.Any(SyntaxKind.PublicKeyword);
```

Comment:
there is a more complex scenario of a public nested class in an internal outer calss (or even some nesting hierarchy).
I think we need to check not only this type but also its containing types that all of them are public.

Reply:
ok

Reply:
@SENya1990 done